src/datasources/asset/AssetDataSource.ts

```diff
@@ -82,4 +85,10 @@ export class AssetDataSource extends DataSourceBase<AssetQuery, AssetDataSourceO
   getListAssetsSource(): ListAssetsDataSource {
     return this.listAssetsDataSource;
   }
+
+  async metricFindQuery(query: AssetQuery): Promise<MetricFindValue[]> {
+    const assetFilter = (query as ListAssetsQuery)?.filter ?? '';
+    const assetsResponse: AssetModel[] = await this.listAssetsDataSource.queryAssets(assetFilter, QUERY_LIMIT);
+    return assetsResponse.map((asset: AssetModel) => ({ text: asset.name, value: `Asset.${asset.vendorName}.${asset.modelName}.${asset.serialNumber}` }));
```

Comment:
The value should not be the id?

Reply:
What happens if 2 assets have the same name?

Reply:
This was brought up already and was a request from @joshuaprewitt  to have it  in this form